### DISCUSSION TITLE
deploy PS script via winget package

### DISCUSSION THREAD

**Discussion by GregoryNimble** [microsoft/winget-cli#1382] [Ideas]
Is deploying PS scripts through winget on the roadmap. This would allow us to deploy custom actions.
If so, what would be the expected timeline?

**Comment by denelon** [comment]
@gorgeron there is at least one similar ask, but I'm not sure this is something we are looking to implement.

https://github.com/microsoft/winget-cli/issues/299

A list of qualities from AppGet influenced our design. They are listed on a blog post about ["winget install learning"](https://devblogs.microsoft.com/commandline/winget-install-learning/) and copied below.

>* No scripts during install – something that we completely agreed with and don’t allow with MSIX
>* Rich manifest definition within GitHub – the power of being open combined with rich declarative meta data about the app is so important to meet goal #&#8203;1 
>* Support all types of Windows applications installers
>* Seamless updates for applications in the repository

**Comment by denelon** [comment]
Can you help me to understand the scenario with a bit more detail? There may be another approach we could take.

**Reply by NimbleTommy** [reply]
We are looking to replace chocolatey with winget, it supports the integration of PowerShell scripts in their packages to allow customization before or after the execution of the installer. This way we are able to configure specific settings that are not provided by the installer.
